packages/expo/cli/utils/analytics/rudderstackClient.ts

```diff
@@ -0,0 +1,99 @@
+import RudderAnalytics from '@expo/rudder-sdk-node';
+import os from 'os';
+import { v4 as uuidv4 } from 'uuid';
+
+import { EXPO_LOCAL, EXPO_STAGING, EXPO_NO_TELEMETRY } from '../env';
+import UserSettings from '../user/UserSettings';
+
+const PLATFORM_TO_ANALYTICS_PLATFORM: { [platform: string]: string } = {
+  darwin: 'Mac',
+  win32: 'Windows',
+  linux: 'Linux',
+};
+
+let client: RudderAnalytics | null = null;
+let identified = false;
+let identifyData: {
+  userId: string;
+  deviceId: string;
+  traits: Record<string, any>;
+} | null = null;
+
+function getClient(): RudderAnalytics {
+  if (client) {
+    return client;
+  }
+
+  client = new RudderAnalytics(
+    EXPO_STAGING || EXPO_LOCAL ? '1wpX20Da4ltFGSXbPFYUL00Chb7' : '1wpXLFxmujq86etH6G6cc90hPcC',
+    'https://cdp.expo.dev/v1/batch',
+    {
+      flushInterval: 300,
+    }
+  );
+
+  // Install flush on exit...
+  process.on('SIGINT', () => client?.flush?.());
+  process.on('SIGTERM', () => client?.flush?.());
+
+  return client;
+}
```

Comment:
```suggestion
function _getUnifiedClient(): RudderAnalytics {
  if (unifiedClient) {
    return unifiedClient;
  }

  unifiedClient = new RudderAnalytics(
     '1wabJGd5IiuF9Q8SGlcI90v8WTs',
    'https://cdp.expo.dev/v1/batch',
    {
      flushInterval: 300,
    }
  );

  // Install flush on exit...
  process.on('SIGINT', () => client?.flush());
  process.on('SIGTERM', () => client?.flush());

  return client;
}
```

Reply:
also, it would be nice to have some async context where we can `await client.flush` on exit.